src/SAML2/Assertion.php

```diff
@@ -164,41 +166,13 @@ class Assertion extends SignedElement
     private $AuthenticatingAuthority = [];
 
     /**
-     * The attributes, as an associative array, indexed by attribute name
+     * The attributes, as an associative array.
      *
-     * To ease handling, all attribute values are represented as an array of values, also for values with a multiplicity
-     * of single. There are 5 possible variants of datatypes for the values: a string, an integer, an array, a
-     * DOMNodeList or a SAML2\XML\saml\NameID object.
-     *
-     * If the attribute is an eduPersonTargetedID, the values will be SAML2\XML\saml\NameID objects.
-     * If the attribute value has an type-definition (xsi:string or xsi:int), the values will be of that type.
-     * If the attribute value contains a nested XML structure, the values will be a DOMNodeList
-     * In all other cases the values are treated as strings
-     *
-     * **WARNING** a DOMNodeList cannot be serialized without data-loss and should be handled explicitly
-     *
-     * @var array multi-dimensional array of \DOMNodeList|\SAML2\XML\saml\NameID|string|int|array
+     * @var array array of attributes indexed by attribute name with each value an SAML2\XML\saml\Attribute 
+     * or an SAML2\XML\saml\NameID object if the attribute is an eduPersonTargetedID
```

Comment:
Maybe we could model this differently?:

```php
     * @var SAML2\XML\saml\Attribute[]
```

I'm also wondering about using the `NameID` here. If the goal is to allow friendly names and stuff like that, we would be missing that part if we use `NameID`s directly. In that case, I think the `NameID` should actually be the value of the `AttributeValue` object. But `AttributeValue`s use DOM objects, so I don't know exactly how to fit both...